system/mp/mpcomm.cpp

```diff
@@ -2178,194 +2265,272 @@ void CMPConnectThread::startPort(unsigned short port)
     if (!listen)
         return;
     running = true;
+    if (acceptThreadPoolSize)
+    {
+        class CFactory : public CInterfaceOf<IThreadFactory>
+        {
+            CMPConnectThread &owner;
+        public:
+            CFactory(CMPConnectThread &_owner) : owner(_owner)
+            {
+            }
+        // IThreadFactory
+            IPooledThread *createNew() override
+            {
+                class CMPConnectionThread : public CInterfaceOf<IPooledThread>
+                {
+                    CMPConnectThread &owner;
+                    Owned<ISocket> sock;
+                public:
+                    CMPConnectionThread(CMPConnectThread &_owner) : owner(_owner)
+                    {
+                    }
+                // IPooledThread
+                    virtual void init(void *param) override
+                    {
+                        sock.set((ISocket *)param);
+                    }
+                    virtual void threadmain() override
+                    {
+                        constexpr unsigned timeoutMs = 5000;
+                        Owned<ISocket> handledSock = sock.getClear();
```

Comment:
I am curious why you move it to a local variable?  To ensure I assume to assume it is detroyed when the thread finishes.  Probably worth a comment.

Reply:
it's because this is a reusable thread.
I want to ensure when the thread is past this threadmain, the socket is no longer a linked member of the underlying reusable thread object.

Arguably, there should be a IPooledThread::cleanup() or similar, which is called after the pooled thread has completed an execution cycle.

I'll add a comment.